core/src/util/secrets.ts

```diff
@@ -46,6 +46,7 @@ export interface Secret {
 }
 
 export type MaybeSecret = string | Secret
+export type MaybeSecretEnv = { [key: string]: MaybeSecret }
```

Comment:
I'm not sure that this type (with this name) is necessary; 
First of all, a key-value map whose values might be secret can be used for more than just env variables.
Second of all, writing `Record<string, MaybeSecret>` is only slightly more verbose but has the same effect and when you see it, you know exactly what it means.

Reply:
I'd be also fine with renaming the type:
```suggestion
export type MaybeSecretRecord = { [key: string]: MaybeSecret }
```